lib/puppet/indirector/face.rb

```diff
@@ -70,9 +70,9 @@ def call_indirection_method(method, key, options)
       if args.length == 1
         key = Puppet[:certname]
       else
-        key = args.first
+        key = args.shift
       end
-      call_indirection_method :find, key, {}
+      call_indirection_method :find, key, args.first
```

Comment:
I was wondering if there are face applications today where `args` contains more than just the `key` and now we'll start passing them through? If so, then I'm thinking we'll fail validation since the argument won't be known? Wondering if maybe we should leave this as-is and override `call_indirection_method` in the `catalog` face? This seems to work:

```ruby
  find.returns <<-'EOT'
    A serialized catalog. When used from the Ruby API, returns a
    Puppet::Resource::Catalog object.
  EOT

  instance_eval do
    def call_indirection_method(method, key, options)
      if method == :find
        facts = Puppet::Face[:facts, '0.0.1'].find(Puppet[:certname])
        super(method, key, facts_for_catalog: facts)
      else
        super(:find, key, {})
      end
    end
  end
```

If we do that, then we don't need to register the `facts_for_catalog` option or modify `face.rb`

Reply:
One problem that arises from this approach is that the time for collection of facts is now embedded in the `retrieval_duration`. Probably not a huge deal, but it will add some time there that wasn't there before.

Reply:
Another option might be to only allow `facts_for_catalog` through at the `face.rb` level, so that we can effectively block all other options.